### ISSUE TITLE
Python highlighting

### ISSUE THREAD

**Issue by Jaymon** [Jaymon/.vim#13]
This doesn't highlight correctly:

```
"{ " + ", ".join(slines) + " }"
```

It highlights everything between`{` and `}` as if it's a substitution.

**Comment by Jaymon** [comment]
```
f"\"{k}\""
```

Also doesn't highlight the `\"` correctly.

**Comment by DonMiller9294** [comment]
The issue you're encountering with the code snippet `"{ " + ", ".join(slines) + " }"` is related to how syntax highlighting works in certain code editors or environments. It seems that the code editor is interpreting everything inside the double curly braces `{}` as a substitution or interpolation, which might not be the intended behavior.

To resolve this issue and have the code properly highlighted, you can escape the curly braces with double curly braces, like this:


"{{ " + ", ".join(slines) + " }}"


By using double curly braces `{{` and `}}`, you are telling the code editor to treat them as literal characters rather than as part of a substitution or interpolation. This should help with the highlighting issue you mentioned.

**Comment by Jaymon** [comment]
```
f"\nfoo bar"
```

The `\n` doesn't highlight correctly either. It looks like f-strings just aren't inheriting something correctly from the normal string highlighting

**Comment by Jaymon** [comment]
@DonMiller9294 yup, I'd prefer not having to change code to get my python to highlight correctly. I don't mind the python highlighter failing on `"{ "` but I'd like it to respect that closing double quote and stop the string formatting highlight so the ` + ", ".join(slines) + ` all highlights correctly.

**Comment by Jaymon** [comment]
```
foo(bar())
```

`foo` is highlighted correctly as `pythonFunctionCall` but `bar` has no highlight class.
